docs_devel/docs/30.CodingStyles.md

```diff
@@ -2,24 +2,51 @@
 
 OmegaT project uses following coding style rules and coding policy.
 
+## Code Style Basics
+
 * Each java file has a GPL3 copyright header. OmegaT has a unit test to check header.
 * Source code has a maximum 120 characters in each line.
 * The Maximum line length of comments in sources is 80 characters 
 * Indentation is 4 spaces. Don't use TAB character.
 * We don't use asterisk import. You should expand all individual imports.
 * You are recommended to put javadoc comments for all your public methods.
 * We don't set method arguments "final", but you are recommended to treat it as final.
+
+## I18N/L10N/Bundles
+
 * Bundle properties should be ASCII, 7bit clean. All translations should be escaped.
-  * It will change to be UTF-8 after OmegaT version 6.1 beta release.
+  - It will change to be UTF-8 after OmegaT version 6.1 beta release.
+
+## API design
+
 * Stable APIs should be under "org.omegat.core" package.
+* Should not remove API, otherwise, please give `@Deprecation` for old API.
+* Please care for backward compatibility, please mind plugin developers.
+
+## Swing GUI design
+
+* We use a Java Swing technology for GUI parts.
 * GUI parts can be;
-  * written by hand, or
-  * Use NetBeans GUI builder
+  - written by hand, or
+  - Use [NetBeans GUI builder](https://netbeans.apache.org/tutorial/main/kb/docs/java/quickstart-gui/)
 * We don't accept other GUI builder materials;
-  * such as JetBrains IntelliJ GUI builder, JForm Designer, Eclipse Window builder.
-  * If you want, you are recommended to implement it as a plugin.
+  - such as JetBrains IntelliJ GUI builder, JForm Designer, Eclipse Window builder.
+  - If you want, you are recommended to implement it as a plugin.
+* We encourage developers to take an approach of MVC design.
+* You can design GUI parts as View, such as `o.o.g.p.v.AppearancePreferencePanel` class.
+* You can write logics as Controller, such as `o.o.g.p.v.AppearanceContoller` class.
+* Sometimes a custom Model is required for complex design, such as `o.o.g.p.v.PluginInfoTableModel`
```

Comment:
It is better to add OmegaT specific operations. For example, we should put localized labels with string got by `OStrings.getString` from bundle. For further maintenance, it is better to make UI component package-private and set its label in Controller using  Mnemonics.setLocalizedText(ViewClass.uiComponentField, OString.getString("TF_LABEL_FOO"));`

Reply:
It can be explained which files define typical UI components 
- Preference pane contents:  classes under `o.o.g.preferences.view` package
- Dialogs: `o.o.g.dialogs`
- aligner: sub-project folder under `aligner/src/main/java` folder
- password dialog for team project:  under `o.o.c.team2.impl` package.

Reply:
Add warning

When UI parts has a comment
```

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    
```
It is generated by NetBeans GUI Editor. You should use the editor to manipulate the file.